src/engine/LocalVocab.cpp

```diff
@@ -100,3 +103,14 @@ bool LocalVocab::isBlankNodeIndexContained(
   }
   return localBlankNodeManager_->containsBlankNodeIndex(blankNodeIndex.get());
 }
+
+// _____________________________________________________________________________
+LocalVocab::Holder LocalVocab::getHolder() const {
+  std::vector<std::shared_ptr<const Set>> wordSets;
```

Comment:
reserve `otherWordSets_.size() + 1`, 
empty shared vocabularies should be a rare case (aren't they filtered out anyway when cloning?, if not that is an easy idea that can be applied in this PR.

Reply:
That's a good point. The other word sets should always be non-empty